Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -167,12 +167,14 @@ public abstract class SinglePassCompiler {
     /// call to the instance Main method in the enclosing class.
     /// </summary>
     protected abstract ConcreteSyntaxTree CreateStaticMain(IClassWriter wr, string argsParameterName);
-    protected abstract ConcreteSyntaxTree CreateModule(string moduleName, bool isDefault, bool isExtern, string/*?*/ libraryName, ConcreteSyntaxTree wr);
+    protected abstract ConcreteSyntaxTree CreateModule(string moduleName, bool isDefault, ModuleDefinition externModule,
```

Comment:
Definitely feels a little weird to have a parameter that is "the module being created, but only passed if it's an extern". Consider keeping `isExtern` and `theModule` as separate parameters.